packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -95,63 +115,168 @@ const getIsReadonlyForNode = (
-              privacy: getPrivacy(typescript, node),
-              readonly: getIsReadonlyForNode(node, analyzer),
+              type: getTypeForNode(statementNode.expression.right, analyzer),
+              privacy: getPrivacy(analyzer.typescript, statementNode),
+              readonly: getIsReadonlyForNode(statementNode, analyzer),
+              ...info,
             })
           );
         }
-      });
-    } else if (typescript.isMethodDeclaration(node) && node.body) {
-      const info = getMemberInfo(typescript, node);
-      const name = node.name.getText();
-      (info.static ? staticMethodMap : methodMap).set(
+      }
+    });
+  }
+};
+
+export const addMethodDeclarationToMethodMap = (
+  node: ts.MethodDeclaration,
+  {methodMap, staticMethodMap}: ClassMemberInfo,
+  analyzer: AnalyzerInterface
+) => {
+  // Ignore non-implementation signatures of overloaded methods by checking
+  // for `node.body`.
+  if (node.body) {
+    const info = getMemberInfo(analyzer.typescript, node);
+    const name = node.name.getText();
+    (info.static ? staticMethodMap : methodMap).set(
+      name,
+      new ClassMethod({
+        ...info,
+        ...getFunctionLikeInfo(node, name, analyzer),
+        ...parseNodeJSDocInfo(node, analyzer),
+      })
+    );
+  }
+};
+
+export const addPropertyDeclarationToFieldMap = (
+  node: ts.PropertyDeclaration,
+  {fieldMap, staticFieldMap}: ClassMemberInfo,
+  analyzer: AnalyzerInterface
+) => {
+  const info = getMemberInfo(analyzer.typescript, node);
+  const map = info.static ? staticFieldMap : fieldMap;
+  map.set(
+    node.name.getText(),
+    new ClassField({
+      ...info,
+      default: node.initializer?.getText(),
+      type: getTypeForNode(node, analyzer),
+      ...parseNodeJSDocInfo(node, analyzer),
+      readonly: getIsReadonlyForNode(node, analyzer),
+    })
+  );
+};
+
+export const addAccessorToAccessorsMap = (
+  node: ts.AccessorDeclaration,
+  {accessorsMap}: ClassMemberInfoIntermediate,
+  analyzer: AnalyzerInterface
+) => {
+  const name = node.name.getText();
+  const _accessors = accessorsMap.get(name) ?? {};
+  if (analyzer.typescript.isGetAccessor(node)) _accessors.get = node;
+  else if (analyzer.typescript.isSetAccessor(node)) _accessors.set = node;
+  accessorsMap.set(name, _accessors);
+};
+
+export const addAccessorsToFieldMap = (
+  {accessorsMap, fieldMap}: ClassMemberInfoIntermediate,
+  analyzer: AnalyzerInterface
+) => {
+  for (const [name, {get, set}] of accessorsMap) {
+    const accessor = get ?? set;
+    if (accessor) {
+      fieldMap.set(
         name,
-        new ClassMethod({
-          ...info,
-          ...getFunctionLikeInfo(node, name, analyzer),
-          ...parseNodeJSDocInfo(node, analyzer),
+        new ClassField({
+          name,
+          type: getTypeForNode(accessor, analyzer),
+          privacy: getPrivacy(analyzer.typescript, accessor),
+          readonly: !!get && !set,
```

Comment:
jsdoc has been missing from accessors: `...parseNodeJSDocInfo(accessor, analyzer)`. Do you mind adding that it so we can obviate my PR at #4198

Reply:
[db95dfd](https://github.com/lit/lit/pull/4167/commits/db95dfdd84c04c5fada8b5b51e474bb4b760438f)